File: BEmuJava/bemu/src/main/java/com/bemu/BEmu/HistoricalDataRequest/MessageHistoric.java
//------------------------------------------------------------------------------
// <copyright project="BEmuJava" file="BEmu.HistoricalDataRequest.MessageHistoric.java" company="Jordan Robinson">
//     Copyright (c) 2013 Jordan Robinson. All rights reserved.
//
//     The use of this software is governed by the Microsoft Public License
//     which is included with this distribution.
// </copyright>
//------------------------------------------------------------------------------

package com.bemu.BEmu.HistoricalDataRequest;

import com.bemu.BEmu.Element;
import com.bemu.BEmu.Name;
import com.bemu.BEmu.Message;
import com.bemu.BEmu.CorrelationID;
import com.bemu.BEmu.Datetime;
import java.util.Map;

public class MessageHistoric extends Message
{
    private final ElementHistoricSecurityData _security;
    
    MessageHistoric(CorrelationID corr, String securityName, Map<Datetime, Map<String, Object>> fieldData, int sequenceNumber)
    {
    	super(new Name("HistoricalDataResponse"), corr, null);
    	this._security = new ElementHistoricSecurityData(securityName, fieldData, sequenceNumber);
    }
    
    ElementHistoricSecurityData security()
    {
    	return this._security;
    }
    
	public Element getElement(String name) throws Exception
	{
		if(name.toLowerCase().equals(this._security.name().toString().toLowerCase()))
			return this._security;
		else
			throw new Exception("not implemented");
	}
    
    public String toString()
    {
        StringBuilder result = new StringBuilder();
        
        result.append("HistoricalDataResponse (choice) = {" + System.getProperty("line.separator"));
        result.append(this._security.prettyPrint(1));
        result.append("}");

        return result.toString();
    }
}

File: BEmu/HistoricalDataRequest/MessageHistoric.cs
//------------------------------------------------------------------------------
// <copyright project="BEmu" file="HistoricalDataRequest/MessageHistoric.cs" company="Jordan Robinson">
//     Copyright (c) 2013 Jordan Robinson. All rights reserved.
//
//     The use of this software is governed by the Microsoft Public License
//     which is included with this distribution.
// </copyright>
//------------------------------------------------------------------------------

namespace BEmu.HistoricalDataRequest
{
    using System;
    using System.Collections.Generic;
    using System.Linq;
    using System.Text;

    internal class MessageHistoric : Message
    {
        private readonly ElementHistoricSecurityData _security;

        internal MessageHistoric(CorrelationID corr, string securityName, List<string> badFields, Dictionary<DateTime, Dictionary<string, object>> fieldData, int sequenceNumber)
            : base(new Name("HistoricalDataResponse"), corr, null)
        {
            this._security = new ElementHistoricSecurityData(securityName, badFields, fieldData, sequenceNumber);
        }

        public override IEnumerable<Element> Elements { get { yield return this._security; } }
        public override object this[string name, int index] { get { return null; } }
        public override BEmu.Element AsElement { get { return new ElementReference(this); } }

        public override string ToString()
        {
            StringBuilder result = new StringBuilder();
            result.Append("HistoricalDataResponse (choice) = {" + Environment.NewLine);
            result.Append(this._security.PrettyPrint(1));
            result.Append("}");

            return result.ToString();
        }

    }
}
